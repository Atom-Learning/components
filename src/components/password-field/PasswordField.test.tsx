import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { axe } from 'jest-axe'
import * as React from 'react'

import { PasswordField } from '.'

describe(`Password component`, () => {
  it('renders a password field', async () => {
    const { container } = render(
      <PasswordField
        name="password"
        css={{ m: 'auto', height: 100, width: 100 }}
      />
    )

    expect(container).toMatchSnapshot()
  })

  it('renders a password field - has no programmatically detectable a11y issues', async () => {
    const { container } = render(<PasswordField name="password" />)

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders with a prompt', async () => {
    const { container } = render(
      <PasswordField
        name="password"
        prompt={{ label: 'Hello', link: '/somewhere' }}
      />
    )

    expect(screen.getByRole('link')).toBeInTheDocument()
    expect(await axe(container)).toHaveNoViolations()
  })

  it('toggles password visibility', async () => {
    render(<PasswordField name="password" />)

    const input = screen.getByLabelText('Password')

    expect(input).toHaveAttribute('type', 'password')

    userEvent.click(screen.getByLabelText('Show password'))
    expect(input).toHaveAttribute('type', 'text')

    userEvent.click(screen.getByLabelText('Hide password'))
    expect(input).toHaveAttribute('type', 'password')
  })
})
