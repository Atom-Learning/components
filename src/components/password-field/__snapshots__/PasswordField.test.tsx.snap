// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password component renders a password field 1`] = `
.sx03kze-fias9 {
  margin: auto;
  height: 100px;
  width: 100px;
  position: relative;
}

.sx03kze-2v396 {
  position: relative;
}

.sx2cs53 {
  display: flex;
}

.sx2cs53-re4mk {
  justify-content: space-between;
  align-items: center;
  margin-bottom: var(--sx-space-2);
}

.sxib16r {
  display: inline-block;
  fill: none;
  stroke: currentcolor;
  stroke-linecap: round;
  stroke-linejoin: round;
  stroke-width: 2;
  vertical-align: middle;
}

.sxib16rmbqqh--size-md {
  height: var(--sx-sizes-2);
  width: var(--sx-sizes-2);
}

.sxdiu5c {
  color: var(--sx-colors-secondary300);
  display: block;
  font-family: var(--sx-fonts-sans);
  font-weight: 500;
  margin: 0;
}

.sxdiu5cjqqvr--size-md {
  font-size: var(--sx-fontSizes-md);
  line-height: 1.625;
}

.sxdiu5cjqqvr--size-md::before {
  content: '';
  margin-bottom: -0.4489em;
  display: table;
}

.sxdiu5cjqqvr--size-md::after {
  content: '';
  margin-top: -0.4489em;
  display: table;
}

.sxy3zvi {
  -webkit-appearance: none;
  appearance: none;
  border: unset;
  border-radius: var(--sx-radii-0);
  cursor: pointer;
  padding: unset;
  transition: all 125ms ease-out;
}

.sxy3zvi2gdaw--size-lg {
  height: var(--sx-sizes-4);
  width: var(--sx-sizes-4);
}

.sxy3zvip7ijy--c2 {
  background: transparent;
  color: var(--sx-colors-tonal700);
}

.sxy3zvip7ijy--c2:hover,
.sxy3zvip7ijy--c2:focus {
  color: var(--sx-colors-primary900);
}

.sxy3zvi-014be {
  bottom: 0;
  position: absolute;
  right: 0;
}

.sx47aq8 {
  -webkit-appearance: none;
  appearance: none;
  border: 1px solid var(--sx-colors-tonal500);
  border-radius: var(--sx-radii-0);
  box-shadow: none;
  box-sizing: border-box;
  color: var(--sx-colors-tonal900);
  cursor: text;
  display: block;
  font-family: var(--sx-fonts-sans);
  font-size: var(--sx-fontSizes-md);
  height: var(--sx-sizes-4);
  width: 100%;
  padding: var(--sx-space-3);
  transition: all 100ms ease-out;
}

.sx47aq8:focus {
  border-color: var(--sx-colors-primary900);
  box-shadow: inset 0 0 0 1px var(--sx-colors-primary900);
  outline: none;
}

.sx47aq8[disabled] {
  background-color: var(--sx-colors-tonal300);
  color: var(--sx-colors-tonal700);
  cursor: not-allowed;
}

<div>
  <form
    aria-label="form"
    class="sx03kze"
  >
    <div
      class="sx03kze sx03kze-fias9"
    >
      <div
        class="sx2cs53 sx2cs53-re4mk"
      >
        <label
          class="sxdiu5c sxdiu5cjqqvr--size-md"
          for="password"
        >
          Password
        </label>
      </div>
      <div
        class="sx03kze sx03kze-2v396"
      >
        <input
          autocomplete="current-password"
          class="sx47aq8"
          id="password"
          name="password"
          type="password"
        />
        <button
          aria-label="Show password"
          class="sxy3zvi sxy3zvi03kze--theme-primary sxy3zvi03kze--appearance-simple sxy3zvi2gdaw--size-lg sxy3zvip7ijy--c2 sxy3zvi-014be"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="sxib16r sxib16rmbqqh--size-md"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 9s-.32.667-1 1.514M12 14a8.478 8.478 0 01-4.051-1M12 14a8.478 8.478 0 004.051-1M12 14v3.5M4 9s.354.737 1.106 1.645M7.95 13L5 16m2.949-3c-1.26-.673-2.198-1.577-2.843-2.355M16.051 13l2.449 3m-2.449-3c1.33-.711 2.303-1.68 2.949-2.486m-13.894.13L2 12m17-1.486L22 12"
            />
          </svg>
        </button>
      </div>
    </div>
  </form>
</div>
`;
